File: refactored/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/LayoutNotEqualsNullIntValueFixedTest.cs
using Xunit;

public class LayoutTypedTests
{
    [Fact]
    public void LayoutNotEqualsNullIntValueFixedTest()
    {
        // Arrange
        int? nullInt = null;
        var layout1 = new Layout<int>("2");
        Layout<int>? layout2 = nullInt;
        // Act + Assert
        Assert.False(layout1 == nullInt, "Explanation message");
        Assert.False(layout1.Equals(nullInt), "Explanation message");
        Assert.False(layout1.Equals((object?)nullInt), "Explanation message");
        Assert.NotEqual(layout1, layout2, "Explanation message");
        Assert.NotEqual(layout1.GetHashCode(), layout2?.GetHashCode(), "Explanation message");
    }
}

File: smelly/assertion-roulette/java-junit/LayoutNotEqualsNullIntValueFixedTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class LayoutTypedTests {
    @Test
    void layoutNotEqualsNullIntValueFixedTest() {
        // Arrange
        Integer nullInt = null;
        Layout<Integer> layout1 = new Layout<>("2");
        Layout<Integer> layout2 = nullInt;
        // Act + Assert
        assertFalse(layout1 == nullInt);
        assertFalse(layout1.equals(nullInt));
        assertFalse(layout1.equals((Object) nullInt));
        assertNotEquals(layout1, layout2);
        assertNotEquals(layout1.hashCode(), layout2.hashCode());
    }
}
